### ISSUE TITLE
[Feature request] Support CancellationTokens on Subscriber handlers

### ISSUE THREAD

**Issue by matt-psaltis** [dotnetcore/CAP#907]
Currently we can pass through CapHeader as a Handle method parameter. CAP picks this up as IsFromCap in the ParameterDescriptor. Can this be extended to CancellationTokens for Async methods? This cancellation token should register with the IBootstrapper.BootstrapAsync cancellation token so that shutdown of the Bootstrapper initiates cancellation of Subscribers.

Example:

```
[Topic("Workshop", typeof(CompanyCreated))]
public Task Handle(CompanyCreated message, CancellationToken cancellationToken) {
}
```

Happy to help with these requests if you're ok with the approach of course :D

**Comment by yang-xiaodong** [comment]
Hello,

This  suggestion looks good, would you like to submit a PR?

**Comment by yang-xiaodong** [comment]
Fixed in `5.1.1-preview-141622241`
